### PR TITLE
shell: Fix and test skip links

### PR DESCRIPTION
**Pull request by martinpitt**
We don't want the skiplinks to *actually* change the URL -- they will go
from e.g. `/system` to `/cockpit/@localhost/shell/index.html#content`.
This causes a page reload, breaks the focusing, and is also not an URL
which we actually want to expose. Fix this by stopping event
progression.

Also remove the `return false` -- event handlers don't return anything.

Introduce a `TestMenu.testKeyboardNavigation` -- skip links and tabbing
was previously not tested at all.

[1] https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#the_event_listener_callback

----

Preparation for #20553 -- let's first test skiplinks before we change them.

### CODE REVIEW

**Comment by martinpitt** [comment]
@marusak As you introduced the skip links, I'd appreciate your review. My interactive test seems quite reasonable now (no reloads and focusing works), but I'd like to make sure it's what you had in mind.

@jelly You recently dabbled with the keyboard test API -- please let me know if `b.key()` works for you or you had something different in mind.

File: test/common/testlib.py [hunk 1]
```diff
@@ -565,6 +565,11 @@ def _key_press_firefox(self, keys: str, modifiers: int = 0, use_ord: bool = Fals
             args["type"] = "keyUp"
             self.cdp.invoke("Input.dispatchKeyEvent", **args)
 
+    # FIXME: This should be key_press(), and key_press() should be something else
+    def key(self, name: str) -> None:
+        self.cdp.invoke("Input.dispatchKeyEvent", type="keyDown", key=name)
+        self.cdp.invoke("Input.dispatchKeyEvent", type="keyUp", key=name)
```
Comment (jelly) [inline test/common/testlib.py:571]: `key` sounds like a good compromise, breaking `key_press` is imo a no-go at this moment as everyone uses it.

Comment (martinpitt) [reply test/common/testlib.py:571]: Yes, I thought about `.key()` and `.input_text()`, moving everything over to these two, and eventually dropping `.key_press()`

**Review by marusak — APPROVED** [review]
Works for me, thanks!

**Pull request merged**